content/documentation/pages/4-batch-developer-guides/2-batch/2-spring-batch.md

```diff
@@ -8,34 +8,310 @@ description: 'Create a Spring Batch Job'
+
+1.  Download the `download: https://raw.githubusercontent.com/spring-cloud/spring-cloud-dataflow-samples/master/dataflow-website/batch-developer-guides/batch/batchsamples/billrun/src/main/resources/schema.sql title=schema.sql` and copy it to the /src/main/resources directory.
+
+1.  In your favorite IDE create the `io.spring.billrun.model` package
+1.  Create a `Usage` class in the `io.spring.billrun.model` using your favorite IDE that looks like the contents in [Usage.java](https://github.com/spring-cloud/spring-cloud-dataflow-samples/tree/master/dataflow-website/batch-developer-guides/batch/batchsamples/billrun/src/main/java/io/spring/billrun/model/Usage.java).
+
+1.  Create a `Bill` class in the `io.spring.billrun.model` using your favorite IDE that looks like the contents in [Bill.java](https://github.com/spring-cloud/spring-cloud-dataflow-samples/tree/master/dataflow-website/batch-developer-guides/batch/batchsamples/billrun/src/main/java/io/spring/billrun/model/Bill.java).
+
+1.  In your favorite IDE create the `io.spring.billrun.configuration` package
+
+1.  Now lets create our `ItemProcessor` for pricing each Usage record. Create a [BillProcessor](https://github.com/spring-cloud/spring-cloud-dataflow-samples/tree/master/dataflow-website/batch-developer-guides/batch/batchsamples/billrun/src/main/java/io/spring/billrun/configuration/BillProcessor.java) class in the `io.spring.billrun.configuration` using your favorite IDE that looks like the contents below.
+
+    ```java
+    public class BillProcessor implements ItemProcessor<Usage, Bill> {
+
+      @Override
+      public Bill process(Usage usage) {
+         Double billAmount = usage.getDataUsage() * .001 + usage.getMinutes() * .01;
+         return new Bill(usage.getId(), usage.getFirstName(), usage.getLastName(),
+               usage.getDataUsage(), usage.getMinutes(), billAmount);
+      }
+    }
+    ```
+
+    Notice that we are implementing the `ItemProcessor` interface that has the `process` method that we need to override.
+    Our parameter is a Usage object and the return value is of type Bill.
+
+1.  Now we will create a Java configuration that will specify the beans required for the BillRun `Job`. In this case create a [BillingConfiguration](https://github.com/spring-cloud/spring-cloud-dataflow-samples/tree/master/dataflow-website/batch-developer-guides/batch/batchsamples/billrun/src/main/java/io/spring/billrun/configuration/BillingConfiguration.java) class in the `io.spring.billrun.configuration` using your favorite IDE that looks like the contents below.
+
+    ```java
+    {/* highlight-range{2-3} */}
+    @Configuration
+    @EnableTask
+    @EnableBatchProcessing
+    public class BillingConfiguration {
```

Comment:
When this class is copied, it does not paste correctly. There is no space between the @EnableBatchProcessing and public for example.